File: app/lib/models/programme_entrainement.dart
import 'groupe_running.dart';

class ProgrammeEntrainement {
  final int id;
  final String titre;
  final String? description;
  final DateTime? dateDebut;
  final DateTime? dateFin;
  final GroupeRunning? groupe;

  ProgrammeEntrainement({
    required this.id,
    required this.titre,
    this.description,
    this.dateDebut,
    this.dateFin,
    this.groupe,
  });

  factory ProgrammeEntrainement.fromJson(Map<String, dynamic> json) {
    return ProgrammeEntrainement(
      id: json['id'] as int,
      titre: json['titre'] as String,
      description: json['description'] as String?,
      dateDebut: json['dateDebut'] != null
          ? DateTime.tryParse(json['dateDebut'].toString())
          : null,
      dateFin: json['dateFin'] != null
          ? DateTime.tryParse(json['dateFin'].toString())
          : null,
      groupe: json['groupe'] != null
          ? GroupeRunning.fromJson(json['groupe'] as Map<String, dynamic>)
          : null,
    );
  }
}

File: demo/src/main/java/com/example/demo/entity/ProgrammeEntrainement.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Table(name = "programmes_entrainement")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProgrammeEntrainement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String titre;

    @Column(columnDefinition = "TEXT")
    private String description;

    private Instant dateDebut;

    private Instant dateFin;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "groupe_id", nullable = false)
    private GroupeRunning groupe;
}
